### ISSUE TITLE
Support for non-serializable types

### ISSUE THREAD

**Issue by follesoe** [follesoe/BlackBoxRecorder#7]
Better exception handling when trying to serialize non-serializable types. Should we provide hooks to register custom serializers for certain types? 

Should also consider using base64 encoding of streams, so that we can do dependency recording on System.IO code.

**Comment by oc** [comment]
I agree we may need a pluggable Conversion registry in the future (atleast in Java).

Streams may be harder to solve in Java, as to my understanding you cannot just peek on an IS or add an ThreadLocal observer straight forwardly (however we could shamelessly dup it) i.e. some call that returns i.e. an IS. In java we could perhaps make use of a library like xstream to serialize harder-to/non-serializated objects. Does .NET have an equivalent?

Did I understand you correctly you mean somethn like this (JSON)
{"klass":"UnserializableClass.class", "unserializable":true, "method": { methodName: "someMethodCall", "params":["param":{"name": "firstParameter", "type": "String", value: "fP" }], "return": { "type":"CustomInputStream", "value": "...the byte[] that was read... i.e. KOI-encoded for the purpose of this example..." }}}  

UnserializableClass unserializableClass = mock(UnserializableClass.class);
when(unserializableClass.someMethodCall("fP")).thenReturn(Convert.<String,CustomInputStream>convert(theStringThatWasReadOrSomething));

That way we could register conversions from <F,T> (F is always string(?)), pseudocodish:

Convert.register(new Conversion<String,CustomInputStream>() { public InputStream convert(final String input) { return new CustomInputStream(base64decode(input.getBytes()), Encoding.newInstance("KOI")); } })
